api/routes.js

```diff
@@ -746,13 +747,18 @@ module.exports = (fastify, opt, next) => {
           logger.info("Empty VAST returned", { label: host });
         } else {
           logger.info("Returned VAST and created a session", { label: req.headers['host'], sessionId: session.sessionId });
+          CloudWatchLog("ADS_RETURNED", req.headers['host'], { dur: session.adBreakDuration, session: session.sessionId });
         }
 
         reply.header("Content-Type", "application/xml; charset=utf-8");
         reply.code(200).send(vast_xml);
       }
     } catch (exc) {
-      logger.error(exc, { label: req.headers['host'], sessionId: session.sessionId });
+      if (session) {
+        logger.error(exc, { label: req.headers['host'], sessionId: session.sessionId });
```

Comment:
not sure if `exc` by itself will let logger.error(...) display the message. 
`logger.error(exc.message, { label: req.headers['host'], sessionId: session.sessionId });`  might be needed.

Reply:
Yes it does. I wanted to send in whatever was caught as it is not always an exception object is of a type containing the `.message`